scripts/generate_zoo_yml.py

```diff
@@ -0,0 +1,57 @@
+import structlog
+
+from zoo.repos.github import create_remote_commit as create_github_remote_commit
+from zoo.repos.github import get_file_content as get_github_file_content
+from zoo.repos.gitlab import create_remote_commit as create_gitlab_remote_commit
+from zoo.repos.gitlab import get_file_content as get_gitlab_file_content
+from zoo.repos.models import Repository
+from zoo.repos.zoo_yml import generate, parse, validate
+from zoo.services.models import Service
+
+log = structlog.get_logger()
+
+
+def file_exists(remote_id, path, provider, ref="master"):
+    get_file_content = None
+    if provider == "github":
+        get_file_content = get_github_file_content
+    else:
+        get_file_content = get_gitlab_file_content
+
+    try:
+        _ = get_file_content(remote_id, path, ref)
```

Comment:
I agree with you that we should almost never catch exceptions this broadly. The reason why this is not done in this specific method is because I still need to catch the exception in the main loop (Service.objects.all()) and continue it if an error occurs. I just say I don't care what error happened, just ignore it and move on. If this was a library or something bigger than a script then hell yeah I would not do this at all.

Reply:
In the django command now file_exists only catches FileNotFoundError